main.go

```diff
@@ -92,15 +91,17 @@ func run(ctx context.Context, env Environment) error {
 			name:     env.DatabaseName,
 			timeout:  env.DatabaseTimeout,
 		},
-		credentials,
+		databaseCredentials,
 	)
 	if err != nil {
 		return fmt.Errorf("unable to connect to database @ %s:%s: %w", env.DatabaseHostname, env.DatabasePort, err)
 	}
 	defer func() {
 		_ = database.Close()
 	}()
-	go vault.RenewDatabaseCredentialsPeriodically(ctx, secret, database.Reconnect) // keep alive
+
+	// lease renewal
+	go vault.PeriodicallyRenewSecrets(ctx, authToken, databaseCredentialsLease, database.Reconnect)
```

Comment:
Do you think PeriodicallyRenewLeases might be a better name for this function since you're using the word lease for all the variable names? (And because "secret" is admittedly a confusing word for new Vault users, when we're not talking about the actual KV secrets they're storing.) 

That would probably remove the need for the comment above this too.

Reply:
I was kind of debating between the two. It's good to have a second a oppinion.